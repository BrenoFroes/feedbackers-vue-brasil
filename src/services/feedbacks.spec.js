import FeedbackService from './feedbacks'
import mockAxios from 'axios'

jest.mock('axios')

describe('FeedbackService', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  it('should return a list of feedbacks', async () => {
    const mockedResponse = { data: {} }
    mockAxios.get.mockImplementationOnce(() => {
      return Promise.resolve(mockedResponse)
    })

    const response = await FeedbackService(mockAxios).getAll()

    expect(mockAxios.get).toHaveBeenCalled()
    expect(response).toMatchSnapshot()
  })
})
