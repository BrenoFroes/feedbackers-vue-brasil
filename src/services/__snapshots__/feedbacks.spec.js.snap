// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedbackService should return a list of feedbacks 1`] = `
Object {
  "data": Object {},
}
`;
